unittests/kv_tests.cpp

```diff
@@ -602,98 +632,167 @@ class kv_tester : public tester {
       // BOOST_TEST(iterlimit({{N(eosio.kvram), 0xFFFFFFFF, false}, {N(eosio.kvdisk), 1, false}}) == "Too many iterators");
    }
 
+   // Make sure that a failed transaction correctly rolls back changes to the database,
```

Comment:
now I see what you are doing. Use my suggestion above, revert these methods and the BOOST_AUTO_TEST_CASE_TEMPLATE will take care of testing both chainbase and rocksdb.

Reply:
or you cans use BOOST_DATA_TEST_CASE_F, but no reason that I can see to repeat all these tests 3 times, unless I am missing something.